pipeline/transform.py

```diff
@@ -1,3 +1,125 @@
+"""Transforms the data by flattening and cleaning before it goes into the load stage"""
 
+import pandas as pd
+from pandas import DataFrame
+
+
+def get_rss_feed_df(file_path: str) -> DataFrame:
+    """Returns the scraped article csv as a DataFrame"""
+
+    df = pd.read_csv(file_path)
+
+    return df
+
+
+def get_scraped_articles_df(file_path: str) -> DataFrame:
+    """Returns the rss_feed csv as a DataFrame"""
+
+    df = pd.read_csv(file_path)
+
+    return df
+
+
+def format_time_to_timestamp(time_in_col: str) -> str:
+    """Formats time to timestamp format to load into postgres"""
+    
+    month_to_number =  {"jan": "01", "feb": "02", "mar": "03",
+                        "apr": "04", "may": "05", "jun": "06",
+                        "jul": "07", "aug": "08", "sep": "09",
+                        "oct": "10", "nov": "11", "dec": "12",
+                        }
+    
+    time_in_col = time_in_col.split(" ")
+    time_in_col = time_in_col[1:]
+
+    time_in_col.insert(0 , time_in_col.pop(2))
+    time_in_col.insert(1 , time_in_col.pop(2))
+
+    time_in_col[1] = month_to_number[time_in_col[1].lower()]
+
+
+    if time_in_col[-1].upper()  == "GMT":
+         date = "-".join(time_in_col[:-2])
+         time = time_in_col[-2]
+
+         return f"{date} {time}"
+    #TODO what if time is not GMT?
+    
+    return
+
+
+def format_rss_feed_df(df: DataFrame) -> DataFrame:
+    """Formats rss feed dataframe columns before saving to csv df"""
+
+    df["published"] = df["published"].apply(format_time_to_timestamp)
+    df["id"] = df["id"].apply(lambda url: url.strip())
+    df = df[["id", "title", "published"]]
+
+    return df
+
+
+def format_authors(authors: str) -> str | None :
+    """Returns a list of authors if they exist, and none if they dont"""
+
+    if authors.lower() == "nan":
+        return None
+    
+    if authors[:3].lower() == "by ":
+        authors = authors.lower().replace("by ", "").replace(" &", ",")
+        authors = authors.replace(" and", ",")
+        authors = authors.split(", ")
+
+        authors = list(map(lambda author: author.title(), authors))
+
+        return authors
+
+
+def get_source(source: str) -> str:
```

Comment:
we would likely do a separate pipeline for different news sources because of internal formatting, so this function isn't necessary

Reply:
The other comparison pipeline doesn't run until after the data is in the db - so i disagree with this :/ (it doesn't hurt to include it)